File: 007_Bitwise_Operators_Modifiers/hw_2.cpp

// WAF to reverse an Integer n.

#include <iostream>
using namespace std;

int reverseNumber(int n) {
    int rev = 0;
    while (n != 0) {
        int digit = n % 10;
        rev = rev * 10 + digit;
        n = n / 10;
    }
    return rev;
}

int main() {
    int n;
    cout << "Enter a number: ";
    cin >> n;

    int reversed = reverseNumber(n);
    cout << "Reversed Number: " << reversed << endl;

    return 0;
}

File: 007_Bitwise_Operators_Modifiers/hw_2.py

def reverse_number(n):
    sign = -1 if n < 0 else 1
    n = abs(n)
    rev = 0
    while n != 0:
        digit = n % 10
        rev = rev * 10 + digit
        n = n // 10
    return sign * rev

# Test
n = int(input("Enter a number: "))
print("Reversed number:", reverse_number(n))